experiment_handler/experiment_handler.py

```diff
@@ -208,24 +209,24 @@ def update_experiment():
             printing("Sending new experiment from beginning")
             # starting anew
             exp.build_scans()
-            pickled_exp = pickle.dumps(exp)
+            serialized_exp = pickle.dumps(exp, protocol=pickle.HIGHEST_PROTOCOL)
```

Comment:
Could use a comment here describing this line since it's not obvoius what highest protocol means.
Suggestion: # Use the newest, fastest pickling protocol (=2 as of python 2.3, see docs).